stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/sync/internal/SyncManager.kt

```diff
@@ -245,86 +250,102 @@ internal class SyncManager(
         }
     }
 
-    private suspend fun retryChannels(): List<Channel> {
-        return repos.selectChannelsSyncNeeded().onEach { channel ->
-            chatClient.createChannel(
+    private suspend fun retryChannels() {
+        val cids = repos.selectChannelCidsSyncNeeded()
```

Comment:
Why not keep the original `selectChannelsSyncNeeded()`? I think we gonna lose a bit of performance by making 2 database calls in this case. It adds a bit more code too.

Reply:
By reading your documentation again. That was for performance reasons, right? If that's the case. I think we can use functional programing instead of a `for` loop... But we would still make database requests in a loop, I'm a bit skeptical about this...

Reply:
Yeah, there was a potential performance issue of reading unlimited records.
Database call is not so significant compareding to server call which happens next.